lib/index.js

```diff
@@ -669,103 +685,103 @@ internals.getJoinTableInfo = (model, knex, cb) => {
 
         const tableName = joinTableFrom[0];
 
-        internals.getColumnsForTable(knex, tableName)
-            .then((dbSchema) => {
-
-                const dbColsToConvert = Object.keys(dbSchema)
-                    .filter((colName) => Mappings.ignoreColumns.indexOf(dbSchema[colName]) === -1);
+        internals.getColumnsForTable(knex, tableName, (err, dbSchema) => {
 
-                const db2JoiTypes = dbColsToConvert.reduce((collector, columnName) => {
-
-                    const currentColumn = dbSchema[columnName];
-                    const db2Joi = Mappings.convertFuncs.knex2JoiType(Mappings.convertFuncs.db2Knex(currentColumn));
+            if (err) {
+                return next(err);
+            }
 
-                    collector[columnName] = db2Joi;
+            const dbColsToConvert = Object.keys(dbSchema)
+                .filter((colName) => {
 
-                    return collector;
-                }, {});
+                    const colType = dbSchema[colName];
+                    const [err, val] = Mappings.convertFuncs.getAliasVal(colType);
+                    if (!!err) {
+                        internals.skippedCols.push(`model: ${tableName}, colName: ${colName}, colType: ${colType}`);
+                        return false;
+                    }
+                    return true;
+                });
 
+            const db2JoiTypes = dbColsToConvert.reduce((collector, columnName) => {
 
-                let joiSchema;
+                const currentColumn = dbSchema[columnName];
 
-                if (relationExtra.isJoi === true) {
+                // dbColsToConvert has filtered out invalid db column types
+                // and fed them to internals.skippedCols, no error will be
+                // produced here
+                const [ignoreErr, knexType] = Mappings.convertFuncs.db2Knex(currentColumn);
 
-                    joiSchema = relationExtra;
-                }
-                else {
-                    joiSchema = Joi.object(relationExtra.reduce((collector, extra) => {
+                // It's mapping.js's job to keep the left side of knexJoiMap
+                // in parity with the right side of columnCompilerKnexMap
+                // No errors should be produced
```

Comment:
If it shouldn't happen but you also want to express that in code, just add an assertion!  You wont have to hit the assertion in code coverage.